Apps/phradar/radar_connector.py

```diff
@@ -0,0 +1,355 @@
+
+        # store the r_text in debug data, it will get dumped in the logs if the action fail
+        action_result.add_debug_data({"r_status_code": resp.status_code})
+        action_result.add_debug_data({"r_text": resp.text})
+        action_result.add_debug_data({"r_headers": resp.headers})
+
+        # Process each "Content-Type" of response separately
+
+        # Process a json response
+        if "json" in resp.headers.get("Content-Type", ""):
+            return self._process_json_response(resp, action_result)
+
+        # Process an HTML response, Do this no matter what the api talks.
+        # There is a high chance of a PROXY in between phantom and the rest of
+        # world, in case of errors, PROXY"s return HTML, this function parses
+        # the error and adds it to the action_result.
+        if "html" in resp.headers.get("Content-Type", ""):
+            return self._process_html_response(resp, action_result)
+
+        # it"s not content-type that is to be parsed, handle an empty response
+        if not resp.text:
+            return self._process_empty_response(resp, action_result)
+
+        # everything else is actually an error at this point
+        message = f"Error from Radar API. Status Code: {resp.status_code} Data from server: {resp.text.replace('{', '{{').replace('}', '}}')}"
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, method="get", **kwargs):
+        self.save_progress(f"make rest call")
+
+        url = f"{self._api_url}{endpoint}"
+        config = self.get_config()
+        resp_json = None
+        request_headers = {
+            "User-Agent": "Splunk Phantom",
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {config['radar_api_token']}"
+        }
+
+        self.save_progress(f"send {method} request to {url}")
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(action_result.set_status(phantom.APP_ERROR, f"Invalid method: {method}"), resp_json)
+
+        try:
+            resp = request_func(url, verify=self._verify_ssl, headers=request_headers, **kwargs)
+            return self._process_response(resp, action_result)
+
+        except Exception as ex:
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR,
+                    f"Error Connecting to server. Details: {str(ex)}"
+                ),
+                resp_json
+            )
+
+    def _handle_test_connectivity(self, param):
+        self.save_progress("connect to Radar")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        radar_val, response = self._make_rest_call("/incidents", action_result, method="get", params=None)
```

Comment:
Yes this will still work even if no incidents are present, however we are considering revising our connectivity test to be something less generic.

Reply:
Sure, we just want to make sure there are not special cases where this would fail in a customer environment simply due to configuration or lack of data.

Reply:
For sure, the only way the current check would fail is if the token they were using had insufficient permissions or was invalid. We talked more about the connectivity test today and think using this endpoint will be reasonable so long as we have adequate documentation and error handling, if your team is ok with that.

Reply:
That seems reasonable to me as it is pretty consistent with other apps.